packages/ember-metal/lib/mixin.ts

```diff
@@ -251,7 +251,7 @@ function mergeMixins(
   descs: { [key: string]: object },
   values: { [key: string]: object },
   base: { [key: string]: object },
-  keys: string[]
+  keys: Set<string>
```

Comment:
changing to `Set` avoids duplicate  keys which manifested itself with following code
```
     let MyMixin = Mixin.create({
        foo: observer('bar', function() {}),
      });

      let Mixin2 = Mixin.create({
        foo: observer('baz', function() {}),
      });

      let obj = mixin({}, MyMixin, Mixin2);
```